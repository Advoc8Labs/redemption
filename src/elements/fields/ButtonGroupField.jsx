import React from 'react';

function ButtonGroupField({
  element,
  applyStyles,
  fieldLabel,
  fieldVal = null,
  onClick = () => {},
  elementProps = {}
}) {
  const servar = element.servar;
  return (
    <>
      {fieldLabel}
      <div
        css={{
          display: 'flex',
          flexWrap: 'wrap',
          width: '100%',
          ...applyStyles.getTarget('fc')
        }}
        {...elementProps}
      >
        {servar.metadata.options.map((opt, index) => {
          const imageUrl = servar.metadata.option_images[index];
          return (
            <div
              id={servar.key}
              onClick={onClick}
              key={opt}
              css={{
                boxSizing: 'border-box',
                display: 'flex',
                justifyContent: 'center',
                alignItems: 'center',
                flexDirection: 'column',
                cursor: 'pointer',
                ...applyStyles.getTarget('field'),
                '&:active': applyStyles.getTarget('active'),
                '&:hover': applyStyles.getTarget('hover'),
                ...(fieldVal === opt ? applyStyles.getTarget('active') : {})
              }}
            >
              {imageUrl && (
                <img
                  src={imageUrl}
                  style={{
                    // Setting min-height to 0 prevents vertical image overflow
                    minHeight: 0,
                    objectFit: 'contain'
                  }}
                />
              )}
              {opt}
            </div>
          );
        })}
      </div>
    </>
  );
}

export default ButtonGroupField;
