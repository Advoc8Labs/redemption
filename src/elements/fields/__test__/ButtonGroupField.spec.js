import React from 'react';
import { create, act } from 'react-test-renderer';
import ButtonGroupField from '../ButtonGroupField';
import ApplyStyles from '../../styles';

describe('ButtonGroupField', () => {
  it('renders an empty Button group', () => {
    // Arrange
    const props = {
      element: {
        id: 'fieldId',
        apply_styles: {},
        styles: {
          height: '50',
          height_unit: '%',
          width: '50',
          width_unit: 'px',
          background_color: '000000',
          uploader_padding_top: '0',
          uploader_padding_bottom: '0',
          uploader_padding_left: '0',
          uploader_padding_right: '0',
          cta_padding_top: '0',
          cta_padding_bottom: '0',
          cta_padding_left: '0',
          cta_padding_right: '0'
        },
        mobile_styles: {},
        servar: {
          key: 'key',
          name: 'Name',
          metadata: {
            options: []
          },
          type: 'select'
        },
        properties: { icon_url: 'https://feathery.tech' }
      },
      required: true,
      onChange: jest.fn(),
      onClick: jest.fn()
    };
    props.applyStyles = new ApplyStyles(props.element, []);

    // Act
    let component;
    act(() => {
      component = create(<ButtonGroupField {...props} />);
    });

    // Assert
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders an empty Button group with options', () => {
    // Arrange
    const props = {
      element: {
        id: 'fieldId',
        apply_styles: {},
        styles: {
          height: '50',
          height_unit: '%',
          width: '50',
          width_unit: 'px',
          background_color: '000000',
          uploader_padding_top: '0',
          uploader_padding_bottom: '0',
          uploader_padding_left: '0',
          uploader_padding_right: '0',
          cta_padding_top: '0',
          cta_padding_bottom: '0',
          cta_padding_left: '0',
          cta_padding_right: '0'
        },
        mobile_styles: {},
        servar: {
          key: 'key',
          name: 'Name',
          metadata: {
            options: ['opt1', 'opt2'],
            option_image_urls: ['', '']
          },
          type: 'select'
        },
        properties: { icon_url: 'https://feathery.tech' }
      },
      required: true,
      onChange: jest.fn(),
      onClick: jest.fn()
    };
    props.applyStyles = new ApplyStyles(props.element, []);

    // Act
    let component;
    act(() => {
      component = create(<ButtonGroupField {...props} />);
    });

    // Assert
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
