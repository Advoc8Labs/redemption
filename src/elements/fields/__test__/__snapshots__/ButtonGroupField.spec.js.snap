// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroupField renders an empty Button group 1`] = `
<div
  className="css-2cev0g-ButtonGroupField"
/>
`;

exports[`ButtonGroupField renders an empty Button group with options 1`] = `
<div
  className="css-2cev0g-ButtonGroupField"
>
  <div
    className="css-1w981eq"
    id="key"
    onClick={[MockFunction]}
  >
    
    opt1
  </div>
  <div
    className="css-1w981eq"
    id="key"
    onClick={[MockFunction]}
  >
    
    opt2
  </div>
</div>
`;
